### PR TITLE
adding changes for issues found and new EP for Roles

### PR DESCRIPTION
**Pull request by JavierMontanoNex**

### CODE REVIEW

File: Support.API.Services/Services/ProfileService.cs [hunk 1]
```diff
@@ -115,6 +115,17 @@ public string CreateUpdateProfile(ProfileRequest data)
...
                     context.SaveChanges();
+
+                    if (!string.IsNullOrEmpty(data.OrganizationId))
+                    {
+                        var org = context.Organizations.Where(x => x.OrganizationId == int.Parse(data.OrganizationId)).FirstOrDefault();
+                        if (org != null)
+                        {
+                            org.IdProfile = profile.ProfileId;
+                            context.Organizations.Update(org);
+                            context.SaveChanges();
```
Comment (rodrigocaballeronexion) [inline Support.API.Services/Services/ProfileService.cs:126]: Let's use SaveChanges as the last action to be called

Comment (JavierMontanoNex) [reply Support.API.Services/Services/ProfileService.cs:126]: this is part of the special cases that we discussed yesterday, this is a IF/ELSE case and the call to SaveChanges is not avoidable in this case.

File: Support.API.Services/Services/IRoleService.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿using Support.API.Services.KoboData;
+using Support.API.Services.Models;
+using Support.API.Services.Models.Request;
+using System.Collections.Generic;
+
+namespace Support.API.Services.Services
+{
+    public interface IRoleService
+    {
+        IEnumerable<RoleResponse> GetAll();
```
[the remark this answers is not in the document]
Comment (JavierMontanoNex) [reply Support.API.Services/Services/IRoleService.cs:10]: ok, changes added.

File: Support.API.Services/Services/RoleService.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    public class RoleService : IRoleService
+    {
+        private readonly ApplicationDbContext applicationDbContext;
+
+        public RoleService(ApplicationDbContext appContext)
+        {
+            this.applicationDbContext = appContext;
+        }
+
+        public Task<List<RoleResponse>> GetAll()
```
Comment (rodrigocaballeronexion) [inline Support.API.Services/Services/RoleService.cs:24]: Change to public async Task<List<RoleResponse>> GetAll()

File: Support.API.Services/Services/RoleService.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+            foreach (Role rol in applicationDbContext.Roles)
+            {
+                list.Add(new RoleResponse()
+                {
+                    RoleId = rol.RoleId.ToString(),
+                    Name = rol.Name
+                });
+            }
+
+            return Task.FromResult(list);
```
Comment (rodrigocaballeronexion) [inline Support.API.Services/Services/RoleService.cs:36]: just return list;

File: Support.API.Services/Services/RoleService.cs [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+        public RoleService(ApplicationDbContext appContext)
+        {
+            this.applicationDbContext = appContext;
+        }
+
+        public Task<List<RoleResponse>> GetAll()
+        {
+            var list = new List<RoleResponse>();
+            foreach (Role rol in applicationDbContext.Roles)
```
Comment (rodrigocaballeronexion) [inline Support.API.Services/Services/RoleService.cs:27]: do a previous line:
var roles = await applicationDbContext.Roles.ToListAsync();
Then do the loop

**Review by rodrigocaballeronexion — CHANGES_REQUESTED** [review]

**Review by rodrigocaballeronexion — APPROVED** [review]

**Pull request merged**